shop/shop-domain/src/test/java/shop/woowasap/shop/domain/product/ProductTest.java

```diff
@@ -0,0 +1,130 @@
+package shop.woowasap.shop.domain.product;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.time.Instant;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullAndEmptySource;
+import org.junit.jupiter.params.provider.ValueSource;
+import shop.woowasap.shop.domain.product.exception.InvalidProductDescriptionException;
+import shop.woowasap.shop.domain.product.exception.InvalidProductNameException;
+import shop.woowasap.shop.domain.product.exception.InvalidProductPriceException;
+import shop.woowasap.shop.domain.product.exception.InvalidProductQuantityException;
+import shop.woowasap.shop.domain.product.exception.InvalidProductSaleTimeException;
+
+@DisplayName("Product 테스트")
+class ProductTest {
+
+    @Nested
+    @DisplayName("Product를 생성 시")
+    class newProductWithBuilder {
+
+        @Test
+        @DisplayName("정상적인 입력인 경우 상품을 생성한다.")
+        void createProduct() {
+            assertDoesNotThrow(() -> Product.builder()
+                .name("name")
+                .description("description")
+                .price("10000")
+                .quantity(1000L)
+                .startTime(Instant.parse("2023-08-05T20:00:00.000Z"))
+                .endTime(Instant.parse("2023-08-05T20:10:00.000Z"))
+                .build()
+            );
+        }
```

Comment:
아래와 같이 given when then 구조로 변경하면 가독성과 여러개의 exception을 잡고있다는 버그가 사라질거 같아요

```suggestion
        void createProduct() {
        	// given
        	String name = "name";
        	String description = "description";
        	String price = "price";
        	long quantity = 1000L;
        	Instant startTime = Instant.parse("2023-08-05T20:00:00.000Z");
        	Instant endTime = Instant.parse("2023-08-05T20:10:00.000Z");
            
            	// when
            	Exception exception = catchException(() -> Product.builder()
            		.name(name)
            		.description(description)
            		.price(price)
            		.quantity(quantity)
            		.startTime(startTime)
            		.endTime(endTime)
            		.build())
            		
            	// then
            	...
        }
```

Reply:
이렇게 하면 좋을 것 같아요! `catchException` 메서드가 `AssertJ` 추가해야해서 일단 추가하고 이렇게 수정해볼게요!